webproject-frontend/src/components/Themes.vue

```diff
@@ -0,0 +1,233 @@
+        <color-picker
+          class="border border-theme-text"
+          v-model="selectedThemeCopy.textColor"
+        ></color-picker>
+      </div>
+
+      <button @click="onSaveActivateClick">Save and Activate</button>
+      <button @click="onSaveClick">Save</button>
+      <button @click="onResetClick">Reset</button>
+    </div>
+  </div>
+</template>
+
+<script>
+import axios from "axios";
+
+export default {
+  name: "Themes",
+  computed: {
+    currentUser() {
+      return this.$store.state.currentUser;
+    }
+  },
+  data() {
+    return {
+      themes: [],
+      selectedTheme: null,
+      selectedThemeCopy: null
+    };
+  },
+  created() {
+    axios
+      .get("/api/themes/available")
+      .then(response => {
+        this.themes = response.data;
+        if (this.themes.length > 0) {
+          this.themes.forEach(theme => {
+            if (theme.id == this.globalTheme.id) {
+              this.selectedTheme = theme;
+              this.selectedThemeCopy = JSON.parse(
+                JSON.stringify(this.selectedTheme)
+              );
+              return;
+            }
+          });
+        }
+      })
+      .catch(error => {});
+  },
+  methods: {
+    onThemeSelectionChange() {
+      this.selectedThemeCopy = JSON.parse(JSON.stringify(this.selectedTheme));
+    },
+    onSaveClick() {
+      this.patchTheme(this.selectedThemeCopy, this.selectedTheme);
+      this.persistSelectedTheme(false);
+    },
+    onSaveActivateClick() {
+      this.patchTheme(this.selectedThemeCopy, this.selectedTheme);
+      this.persistSelectedTheme(true);
+    },
+    onResetClick() {
+      this.selectedThemeCopy = JSON.parse(JSON.stringify(this.selectedTheme));
+    },
+    onDuplicateClick() {
+      let newTheme = JSON.parse(JSON.stringify(this.selectedTheme));
+      newTheme.id = null;
+      newTheme.name += " Copy";
+      newTheme.preset = false;
+      this.themes.push(newTheme);
+      this.selectedTheme = newTheme;
+      this.selectedThemeCopy = JSON.parse(JSON.stringify(this.selectedTheme));
+    },
+    patchTheme(source, target) {
+      target.name = source.name;
+      target.backgroundColor = source.backgroundColor;
+      target.backgroundHighlightColor = source.backgroundHighlightColor;
+      target.primaryColor = source.primaryColor;
+      target.secondaryColor = source.secondaryColor;
+      target.textColor = source.textColor;
+    },
+    persistSelectedTheme(activate) {
+      const config = { headers: { "Content-Type": "application/json" } };
+      if (this.selectedThemeCopy.id != null) {
+        // update theme
+        axios
+          .put(
+            "/api/themes/" + this.selectedTheme.id,
+            this.selectedTheme,
+            config
+          )
+          .then(response => {
+            console.log("Theme saved successfully");
+            if (activate) {
+              this.onActivateClick();
+            }
+          })
+          .catch(error => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/gmasil/Webproject-NG/pullRequest?prid=8166316)
```suggestion
          .catch((error) => {
```